controllers/hyperconverged/hyperconverged_controller.go

```diff
@@ -1330,6 +1340,66 @@ func (r *ReconcileHyperConverged) deleteObj(req *common.HcoRequest, obj client.O
 	return removed, nil
 }
 
+func (r *ReconcileHyperConverged) EvaluateUpgradeEligibility(req *common.HcoRequest) error {
+	podList := &corev1.PodList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(req.Namespace),
+		client.MatchingLabels{"kubevirt.io": "virt-controller"},
+	}
+
+	if err := r.client.List(req.Ctx, podList, listOpts...); err != nil {
+		req.Logger.Info("Failed to list virt-controller pods", "namespace", req.Namespace, "error", err)
+		return fmt.Errorf("failed to list virt-controller pods: %w", err)
+	}
+
+	if len(podList.Items) == 0 {
+		req.Logger.Info("No virt-controller pods found", "namespace", req.Namespace)
+		return nil
+	}
+
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	for _, pod := range podList.Items {
+		if pod.Status.PodIP == "" {
+			continue
+		}
+		metricsURL := fmt.Sprintf("https://%s:%d/metrics", pod.Status.PodIP, 8443)
+
+		resp, err := httpClient.Get(metricsURL)
+		if err != nil {
+			req.Logger.Info("Failed to query metrics from pod", "pod", pod.Name, "error", err)
+			continue
+		}
+
+		if resp.StatusCode != http.StatusOK {
+			req.Logger.Info("Metrics endpoint returned non-200 status", "pod", pod.Name, "status", resp.StatusCode)
+			resp.Body.Close()
+			continue
+		}
+
+		body, err := io.ReadAll(resp.Body)
```

Comment:
I think it's better to use a scanner. The body can be big.

You can do something like
```go
scanner := bufio.NewScanner(resp.Body)
...
for scanner.Scan() {
	if rhel8Regex.MatchString(scanner.Text()) {
		req.Logger.Info("Detected outdated machine type in metrics", "pod", pod.Name, "matched", rhel8Regex.FindString(virtMetricsData))
		req.Upgradeable = false
		return nil
	}
}
```

Reply:
fixed